EKF/ekf_helper.cpp

```diff
@@ -972,12 +972,62 @@ void Ekf::get_gyro_bias(float bias[3])
 	memcpy(bias, temp, 3 * sizeof(float));
 }
 
+// get the full covariance matrix
+const matrix::SquareMatrix<float, 24> Ekf::covariances() const
+{
+	matrix::SquareMatrix<float, 24> cov;
+	for (unsigned n = 0; n < _k_num_states; n++) {
+		for (unsigned m = 0; m < _k_num_states; m++) {
+			cov(n, m) = P[n][m];
+		}
+	}
+	return cov;
+}
+
 // get the diagonal elements of the covariance matrix
-void Ekf::get_covariances(float *covariances)
+const matrix::Vector<float, 24> Ekf::covariances_diagonal() const
 {
+	matrix::Vector<float, 24> diag;
 	for (unsigned i = 0; i < _k_num_states; i++) {
-		covariances[i] = P[i][i];
+		diag(i) = P[i][i];
+	}
+	return diag;
+}
+
+// get the position covariances
+const matrix::SquareMatrix<float, 3> Ekf::position_covariances() const
+{
+	matrix::SquareMatrix<float, 3> cov;
+	for (unsigned n = 0; n < 3; n++) {
+		for (unsigned m = 0; m < 3; m++) {
+			cov(n, m) = P[n + 7][m + 7];
+		}
+	}
+	return cov;
+}
+
+// get the quaternion covariances
+const matrix::SquareMatrix<float, 4> Ekf::quaternion_covariances() const
```

Comment:
I'd rather call this `orientation_covariances` to be consistent with the other naming.

Reply:
alright. done in 9961b59